### ISSUE TITLE
Handling of Relay connections

### ISSUE THREAD

**Issue by skanev** [apollographql/apollo-ios#145]
Hey.

It will be really good if the client has a way to handle connections. As far I understand it, if I want to render an infinite scroll table view that is fed via a connection, I have to handle this completely in my code. I'm uncertain how that interacts with the cache (I assume it doesn't). It feels like a generic enough case to deserve some support in the client.

I've asked @martijnwalraven about this in the GraphQL Slack's #swift channel and he suggested I open an issue. I'd be happy to attempt a pull request (and figure out the code), but I certainly need some architectural guidelines on how the API should look and behave. @martijnwalraven also mentioned that this would be a good place to discuss how this change should look.

**Comment by ocrickard** [comment]
Would love to see this added. This is core to scrolling list views.

**Comment by martijnwalraven** [comment]
For pagination support in general, I'd like to stay as close as possible to [Apollo JS](http://dev.apollodata.com/react/pagination.html), because something like `fetchMore` doesn't tie us to Relay connections specifically.

There are a few things I think we can do to make dealing with those more convenient however, both in the generated code and in the runtime:
- Adding conformance to a `Connection` protocol to the generated types would give us a way to treat connections generically and add extension methods.
- I'd also like to add support for the `@connection` directive that both [Relay](https://facebook.github.io/relay/docs/pagination-container.html#connection-directive) and [Apollo JS](http://dev.apollodata.com/react/pagination.html#connection-directive) support to generate a stable caching key that ignores pagination parameters.

**Comment by crabman448** [comment]
This functionality would be extremely useful 🙌

**Comment by tivadargyorgy** [comment]
Any news about this topic? 
Is there any way to implement the pagination functionality?

**Comment by LeeWong** [comment]
Any news about this topic?

**Comment by designatednerd** [comment]
This is definitely something we should support, especially for larger queries.

**Comment by designatednerd** [comment]
Throwing this link on here as a note to self and for anyone curious about what this is all about: https://blog.apollographql.com/understanding-pagination-rest-graphql-and-relay-b10f835549e7

**Comment by kimdv** [comment]
This might also solve the problem for watching paginated queries

https://spectrum.chat/apollo/apollo-ios/pagination-watchquery~3e3a3a93-cb84-4962-8573-7d683a09f1d4

https://blog.apollographql.com/pagination-and-infinite-scrolling-in-apollo-client-59ff064aac61

**Comment by skanev** [comment]
Hopefully nobody minds, but I'm gonna close this issue.

**Comment by wow-such-amazing** [comment]
@skanev Hey Stefan! What's the decision about this ticket?

**Comment by skanev** [comment]
I don't know, as I'm not a member of the Apollo team, but it has been open for 5 years now, and I have been clearing my open issues in GitHub. Hence why I closed it :)

When I opened it initially, I didn't understand how to use Apollo or the client side cache that well. Since then, I've gained better understanding, I've managed to solve all the problems I've encountered so far, including the one that prompted me to open this issue in the first place.

Specifically, since opening it, I've actually understood how the cache works and how to think about it. Ironically enough, I managed to grok when I did some Apollo in React. Then I just carried the learning. Specifically, I didn't fully comprehend that (1) the client code gets notified when a change in the cache happens, so it could rerender and (2) I don't need `fetchMore`, I could just fetch the next page the traditional way, append it to the connection in the cache, and everything continues to work. That's how I solved infinite scrolling in React, and then I could do the same in iOS. The only difference was that apollo.js had the `fetchMore` convenience, while in the iOS client I kinda had to do it myself. Which is no biggie.

Long story short: I didn't properly appreciate that there was already a solution to my problem without extending abstractions in Apollo iOS. Now I do :)

Hope that makes sense, and apologies for (1) not being more clear when I closed the issue and (2) not understanding Apollo enough when I opened it in the first place 5 years ago ;)

**Comment by wow-such-amazing** [comment]
@skanev That's a very comprehensive response, thank you! Indeed I have confused you to be a member of Apollo Team. My bad, didn't have a look at the creator of the issue 🤪
